snapcraft/internal/sources.py

```diff
@@ -389,7 +389,13 @@ def get(sourcedir, builddir, options):
     handler_class = _get_source_handler(source_type, options.source)
     handler = handler_class(options.source, sourcedir, source_tag,
                             source_branch)
-    handler.pull()
+    try:
+        handler.pull()
+    except FileNotFoundError:
```

Comment:
This doesn't look good. To do it right, the check sould come from the source class.
Like which bzr || raise exception.